### ISSUE TITLE
no function or associated item named `id` found for struct `whirlpool_cpi::state::Whirlpool` in the current scope

### ISSUE THREAD

**Issue by JohnnnyWang** [orca-so/whirlpool-cpi#2]
Dear developer, we have a compilation error when using this library:

```bash
error[E0599]: no function or associated item named `id` found for struct `whirlpool_cpi::state::Whirlpool` in the current scope
--> programs/learn_whirlpool/src/instructions/utils/swap.rs:7:10
|
7 | #[derive(Accounts)]
| ^^^^^^^^ function or associated item not found in `Whirlpool`
|
= note: this error originates in the derive macro `Accounts` (in Nightly builds, run with -Z macro-backtrace for more info)
```
This error occurs during anchor build (specifically, it should occur during the second round of idl compilation)

---

Other information that may be useful:
- The anchor version we use is: 0.30.1
- solana-cli version: 1.18.17
- 
We are using this example for development:
https://github.com/orca-so/whirlpool-cpi-sample/blob/main/anchor-0.29.0/programs/whirlpool-cpi-sample/src/instructions/proxy_swap.rs

Below is our code (we found that the comment: 
pub whirlpool_program: Program<'info, whirlpool_cpi::program::Whirlpool> can eliminate the error)
```rust
#[derive(Accounts)]
pub struct ProxySwap<'info> {

    // Comment the following program account will eliminate compilation errors.
    pub whirlpool_program: Program<'info, whirlpool_cpi::program::Whirlpool>,

    #[account(address = token::ID)]
    pub token_program: Program<'info, Token>,

    pub token_authority: Signer<'info>,

    #[account(mut)]
    pub whirlpool: Box<Account<'info, Whirlpool>>,

    #[account(mut, constraint = token_owner_account_a.mint == whirlpool.token_mint_a)]
    pub token_owner_account_a: Box<Account<'info, TokenAccount>>,
    #[account(mut, address = whirlpool.token_vault_a)]
    pub token_vault_a: Box<Account<'info, TokenAccount>>,

    #[account(mut, constraint = token_owner_account_b.mint == whirlpool.token_mint_b)]
    pub token_owner_account_b: Box<Account<'info, TokenAccount>>,
    #[account(mut, address = whirlpool.token_vault_b)]
    pub token_vault_b: Box<Account<'info, TokenAccount>>,

    #[account(mut, has_one = whirlpool)]
    pub tick_array_0: AccountLoader<'info, TickArray>,

    #[account(mut, has_one = whirlpool)]
    pub tick_array_1: AccountLoader<'info, TickArray>,

    #[account(mut, has_one = whirlpool)]
    pub tick_array_2: AccountLoader<'info, TickArray>,

    #[account(
        mut, seeds = [b"oracle", whirlpool.key().as_ref()], bump, seeds::program = whirlpool_program.key()
    )]
    pub oracle: UncheckedAccount<'info>,
}
```

**Comment by 0x-skuld** [comment]
Found a solution yet?

**Comment by acheroncrypto** [comment]
This was a bug in Anchor that got fixed recently (https://github.com/coral-xyz/anchor/pull/3228).

Problem happens because of the first account:

```rs
pub whirlpool_program: Program<'info, whirlpool_cpi::program::Whirlpool>,
```

It expects `Whirlpool` rather than the full path. You can add a `use` statement as a workaround.

**Comment by yugure-orca** [comment]
@acheroncrypto 
Thank you!
You saved me and users 😻

Fixed sample code for 0.30.1.
